WordPress/Classes/ViewRelated/Post/AbstractPostListViewController.swift

```diff
@@ -805,8 +805,8 @@ class AbstractPostListViewController: UIViewController, WPContentSyncHelperDeleg
         WPAnalytics.track(.postListPublishAction, withProperties: propertiesForAnalytics())
 
         apost.status = PostStatusPublish
-        if let date = apost.dateCreated(), date == (Date() as NSDate).laterDate(date) {
-            apost.setDateCreated(Date())
+        if let date = apost.dateCreated, date == (Date() as NSDate).laterDate(date) {
```

Comment:
IMHO the code here is also suffering from handling details that should be abstracted from it.

Think of switching this to something like:

```swift
apost.dateCreated = getMeMyDate() // Name is generic on purpose for the example
```

This would put all the date selection logic inside a single method that could then be properly documented.